src/UrlHandlers/CollectionUrlHandling.php

```diff
@@ -53,7 +53,7 @@ protected function getMatchingUrlFragment(Request $request, $currentUrlFragment
 
         $this->matchedUrl = $this->url;
 
-        if (preg_match("|^" . $this->url . "/?([[:digit:]]+)/?|", $uri, $match)) {
+        if (preg_match("`^" . $this->url . "/?([0-9]+)(/|$)`", $uri, $match) && isset($match[1])) {
```

Comment:
One issue might be that this allows for /contacts/3 and /contacts/3/

That's very often an issue as navigation can use relative urls e.g. ../ to go back up to the contacts list. However if someone crafts the link wrong (/contacts/3) the page appears properly but the navigation links are all screwed up. Would it be safer to just not match until the leading slash is on there to remind people about that?